solver/src/main/java/org/chocosolver/solver/constraints/binary/PropPowEven.java

```diff
@@ -91,14 +76,16 @@ void setBounds() throws ContradictionException {
     }
 
     int floor_nroot(int n) {
-        if (n < 0)
+        if (n < 0) {
             return 0;
+        }
         return MathUtils.safeCast((long) Math.floor(Math.pow(n, 1. / exponent)));
```

Comment:
I wonder if Math.sqrt(n) has not a more efficient implementation than Math.pow(n, 1. / 2)

Reply:
I just tested it and it seems to be the case. That said, given our use-case, I don't think it really matters (especially to maintain a propagator just for this). But I can add an if on the exponent, and call Math.sqrt whenever exponent = 2, if you want.

FYI, to compute all square root of integers from 0 to 100 000 000, it takes ~1100 ms on my computer with this formula, and ~1 ms with Math.sqrt.

Reply:
interesting! I am a bit surprised by the 1ms to compute 100 000 000 things but anyway I would normally suggest to add an IF to catch the case exp = 2. 

However, let's wait the answer of @cprudhom regarding the PRECISION. Because if precision is required, then we should probably stick to the pow